src/test/java/Entities/IngredientTest.java

```diff
@@ -12,15 +16,46 @@
     @Test
     public void testName() {
         Ingredient ingredient = new Ingredient("my name", Collections.emptyList());
-
         assertEquals("my name", ingredient.name());
     }
 
     @Test
-    public void testTags() {
+    public void testTagsEmpty() {
         Ingredient ingredient = new Ingredient("", Collections.emptyList());
-
         assertEquals(0, ingredient.tags().size());
     }
 
+    @Test
+    public void testTagsValues() {
+        Tag tag1 = new Tag("Dairy");
+        Tag tag2 = new Tag("Gluten");
+
+        List<Tag> list = new ArrayList<Tag>();
+        list.add(tag1);
+        list.add(tag2);
+
+        Ingredient ingredient = new Ingredient("name", list); // ingredient containing tags now
+
+        assertEquals(list, ingredient.tags());
+
+    }
+
+    @Test
+    public void testHasTag() {
+        Tag tag1 = new Tag("Dairy");
+        Tag tag2 = new Tag("Gluten");
+        Tag tag3 = new Tag("Sugar");
+
+        List<Tag> list = new ArrayList<Tag>();
+        list.add(tag1);
+        list.add(tag2);
+
+        Ingredient ingredient = new Ingredient("name", list); // ingredient containing tags now
+
+        assertEquals(true, ingredient.has(tag1));
+        assertEquals(false, ingredient.has(tag3));
```

Comment:
Not sure if this is true, but there may just be an `assertTrue`/`assertFalse` function so you don't have to pass in the true or false. Google that (junit assert true).